api/modules/users/views.py

```diff
@@ -166,7 +166,7 @@ def trip_friends_all(request):
     except Exception as e:
         return Response(str(e), status=status.HTTP_400_BAD_REQUEST)
 
-    serializer = UserSerializer(all_trips, many=True)
+    serializer = UserOnlySerializer(all_trips, many=True)
```

Comment:
You need to add a check if the current signed in user is present in `all_trips` then remove it from the list
[Reference](https://github.com/project-travel-mate/server/blob/master/api/modules/trips/views.py#L60)